ui-grid/ui-grid.d.ts

```diff
@@ -3548,8 +3548,13 @@ declare namespace uiGrid {
         name?: string;
         /** Sort on this column */
         sort?: ISortInfo;
-        /** Algorithm to use for sorting this column. Takes 'a' and 'b' parameters like any normal sorting function. */
-        sortingAlgorithm?: (a: any, b: any) => number;
+        /**
+         * Algorithm to use for sorting this column. Takes 'a' and 'b' parameters
+         * like any normal sorting function with additional 'rowA', 'rowB', and 'direction'
+         * parameters that are the row objects and the current direction of the sort
+         * respectively. 
+         */
+        sortingAlgorithm?: (a: any, b: any, rowA: IGridRowOf<TEntity>, rowB: IGridRowOf<TEntity>, direction: string) => number;
```

Comment:
I don't think they should be optional.

If they were optional then ui-grid would be able to not provide values for them when calling the callback, and the documentation doesn't say anything about that.

Note that the function you assign to `sortingAlgorithm` does not need to declare all parameters - it could just declare `a` and `b` and ignore the others.